tests/test_timefuncs.py

```diff
@@ -1035,3 +1038,18 @@ def test_timedelta_range() -> None:
 def test_dateoffset_freqstr() -> None:
     offset = DateOffset(minutes=10)
     check(assert_type(offset.freqstr, str), str)
+
+
+def timedelta64_and_arithmatic_operator() -> None:
+    s1 = pd.Series(data=pd.date_range("1/1/2020", "2/1/2020"))
+    s2 = pd.Series(data=pd.date_range("1/1/2021", "2/1/2021"))
+    s3 = s2 - s1
+    td = np.timedelta64(1, "M")
+    check(assert_type((s1 - td), TimestampSeries), TimestampSeries, pd.Timestamp)
```

Comment:
```
check(assert_type((s1 - td), "TimestampSeries"), pd.Series, pd.Timestamp)
```
All the tests should look like this.  The type `TimestampSeries` isn't known at runtime, so you have to surround it with quotes.  

you will have to do something similar for `TimedeltaSeries` and `pd.Series[float]`. 

Look elsewhere in the code to see examples

Reply:
Should we also change thte left 3 tests because they cannot pass pytets

Reply:
Good point.  Move those 3 tests to the end and do
```python
if TYPE_CHECKING_INVALID_USAGE:
     # test 1 here
     # test 2 here
     # test 3 here
```
you import `TYPE_CHECKING_INVALID_USAGE` from `tests`